plugin/Template/TemplateCall.php

```diff
@@ -94,39 +94,52 @@ public function loadTemplates()
-
-		$plugin_name = basename(plugin_dir_path(ONOFFICE_PLUGIN_DIR . '/index.php'));
-		foreach ($templatesAll as $filePath) {
-			$fileName = substr(strrchr($filePath, "/"), 1);
-			if (strpos($filePath, 'themes') !== false) {
-				$filePath = __String::getNew($filePath)->replace(get_template_directory() . '/', '');
-				$templateTitle = 'Personalized (Theme)';
-				$shortPath = '/onoffice-theme/templates/' . $directory . '/';
+		$hasTemplateFolder  = false;
+		$hasPluginFolder    = false;
+		$plugin_name        = basename( plugin_dir_path( ONOFFICE_PLUGIN_DIR . '/index.php' ) );
+
+		foreach ( $templatesAll as $filePath ) {
+			$fileName = substr( strrchr( $filePath, "/" ), 1 );
+			if ( strpos( $filePath, 'themes' ) !== false ) {
+				$hasTemplateFolder = true;
+				$filePath          = __String::getNew( $filePath )->replace( get_template_directory() . '/', '' );
+				$templateTitle     = 'Personalized (Theme)';
+				$shortPath         = '/onoffice-theme/templates/' . $directory . '/';
 			} else {
-				$filePath = __String::getNew($filePath)->replace(plugin_dir_path(ONOFFICE_PLUGIN_DIR), '');
-				if (strpos($filePath, 'onoffice-personalized') !== false) {
-					$templateTitle = 'Personalized (Plugin)';
-					$shortPath = 'onoffice-personalized/templates/' . $directory . '/';
+				$filePath = __String::getNew( $filePath )->replace( plugin_dir_path( ONOFFICE_PLUGIN_DIR ), '' );
+				if ( strpos( $filePath, 'onoffice-personalized' ) !== false ) {
+					$hasPluginFolder = true;
+					$templateTitle   = 'Personalized (Plugin)';
+					$shortPath       = 'onoffice-personalized/templates/' . $directory . '/';
 				} else {
-					$templateTitle = 'Included';
-					$shortPath = $plugin_name . '/' . 'templates.dist/' . $directory . '/';
+					$templateTitle = 'Default';
+					$shortPath     = $plugin_name . '/' . 'templates.dist/' . $directory . '/';
 				}
 			}
-			$folderOrder = self::ORDER_OF_TEMPLATES_FOLDER[$templateTitle];
-			$templatePathGroupByFolder[$templateTitle][$filePath] = $fileName;
+			$folderOrder                                              = self::ORDER_OF_TEMPLATES_FOLDER[ $templateTitle ];
+			$templatePathGroupByFolder[ $templateTitle ][ $filePath ] = $fileName;
 
-			$templateFolderData[$folderOrder]['path'] = $templatePathGroupByFolder[$templateTitle];
-			$templateFolderData[$folderOrder]['title'] = $templateTitle;
-			$templateFolderData[$folderOrder]['folder'] = $shortPath;
+			$templateFolderData[ $folderOrder ]['path']   = $templatePathGroupByFolder[ $templateTitle ];
+			$templateFolderData[ $folderOrder ]['title']  = $templateTitle;
+			$templateFolderData[ $folderOrder ]['folder'] = $shortPath;
+		}
+		if ( $hasTemplateFolder !== $hasPluginFolder ) {
+			foreach ( $templateFolderData as $key => $templateData ) {
+				if ( strpos( $templateData['title'], "Personalized" ) !== false ) {
+					$templateFolderData[ $key ]['title'] = "Personalized";
+				}
+			}
 		}
+		ksort( $templateFolderData );
```

Comment:
Can this be simplified?

For example, you [merge](https://github.com/onOfficeGmbH/oo-wp-plugin/pull/114/files/639c4d0e062eb263c57b65ae0956266d087da159..c8ee5df2468b3de82ced360529efbdf07e053fbc#diff-b77363609ca4a4f37d1ecd149c1d034d6ff43a880bd44945099368e1762a72d9R123) all the paths together before passing them in as `$templatesAll`, but then in this function you separate them again.

So it seems that maybe you can keep the template locations separated so that you avoid the need to separate them in this function.

I hope this comment is understandable. If you have questions, feel free to ask here. :)

Reply:
Oh, nice idea, I'll try it.